lib/engine/game/g_1860.rb

```diff
@@ -252,11 +320,16 @@ def current_layer
       end
 
       def float_corporation(corporation)
-        # Designer says that bankrupt corps keep insolvency flag
-        clear_bankrupt(corporation)
+        clear_bankrupt!(corporation)
         super
       end
 
+      def action_processed(_action)
+        @corporations.each do |corporation|
```

Comment:
you need to do this on every single action? that's gonna make the action * N

Reply:
This is already in the base class, and if anything, mine is simpler:

```
      def action_processed(_action)
        @corporations.dup.each do |corporation|
          close_corporation(corporation) if corporation.share_price&.type == :close
        end if stock_market.has_close_cell
      end
```